### DISCUSSION TITLE
Any example about implicit definition of state space and action space?

### DISCUSSION THREAD

**Discussion by NeroBlackstone** [JuliaPOMDP/POMDPs.jl#436] [Problem Implementation] [answered]
[http://juliapomdp.github.io/POMDPs.jl/stable/def_pomdp/#space_representation](http://juliapomdp.github.io/POMDPs.jl/stable/def_pomdp/#space_representation)

"However it should be noted that, for many solvers, an explicit enumeration of the state and observation spaces is not needed."

How to define state space and action space implicitly? Could you give us some code examples using QuickMDP function?

**Comment by zsunberg** [comment, marked as answer]
Hi @NeroBlackstone two examples are the observation and state space in the partially observable mountaincar example. http://juliapomdp.github.io/POMDPs.jl/stable/def_pomdp/#po-mountaincar

There is no `states` keyword argument, but the type of the state is inferred from `initialstate`. Similarly, there is no explicit definition of the observation space via the `observations` keyword argument, but the type is inferred from the `obstype` argument.
